### PR TITLE
validating date on application submission

### PR DESCRIPTION
**Pull request by pkgodara**

### CODE REVIEW

File: includes/AppLeave.php [hunk 1]
```diff
@@ -75,6 +75,27 @@
 	require_once "../LocalSettings.php";
 	require_once "Globals.php";
 	
+	// validate dates
```
Comment (pkgodara) [inline includes/AppLeave.php:78]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 2]
```diff
@@ -75,6 +75,27 @@
...
+	//
```
Comment (pkgodara) [inline includes/AppLeave.php:79]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 3]
```diff
@@ -75,6 +75,27 @@
...
+	$dtnow = date('Y-m-d');
```
Comment (pkgodara) [inline includes/AppLeave.php:80]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 4]
```diff
@@ -75,6 +75,27 @@
...
+	if( isset( $_POST['Start_Date'] ) )
```
Comment (pkgodara) [inline includes/AppLeave.php:81]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed
* Expected 1 space after IF keyword; 0 found
* Expected 0 spaces after opening bracket; 1 found
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found
* Expected 1 space after closing parenthesis; found 5

File: includes/AppLeave.php [hunk 5]
```diff
@@ -75,6 +75,27 @@
...
+	{
```
Comment (pkgodara) [inline includes/AppLeave.php:82]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 6]
```diff
@@ -75,6 +75,27 @@
...
+		if( strtotime($_POST['Start_Date']) < strtotime($dtnow) )
```
Comment (pkgodara) [inline includes/AppLeave.php:83]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed
* Expected 1 space after IF keyword; 0 found
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found
* Expected 1 space after closing parenthesis; found 9

File: includes/AppLeave.php [hunk 7]
```diff
@@ -75,6 +75,27 @@
...
+		{
```
Comment (pkgodara) [inline includes/AppLeave.php:84]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 8]
```diff
@@ -75,6 +75,27 @@
...
+			echo "ERROR: Start Date cannot be less than Current Date";
```
Comment (pkgodara) [inline includes/AppLeave.php:85]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 9]
```diff
@@ -75,6 +75,27 @@
...
+			die();
```
Comment (pkgodara) [inline includes/AppLeave.php:86]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 10]
```diff
@@ -75,6 +75,27 @@
...
+		}
```
Comment (pkgodara) [inline includes/AppLeave.php:87]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 11]
```diff
@@ -75,6 +75,27 @@
...
+	}
```
Comment (pkgodara) [inline includes/AppLeave.php:88]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 12]
```diff
@@ -75,6 +75,27 @@
...
+
```
Comment (pkgodara) [inline includes/AppLeave.php:89]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 13]
```diff
@@ -75,6 +75,27 @@
...
+	if( isset( $_POST['Start_Date'] ) )
+	{
+		if( strtotime($_POST['Start_Date']) < strtotime($dtnow) )
+		{
+			echo "ERROR: Start Date cannot be less than Current Date";
+			die();
+		}
+	}
+	
+	if( isset( $_POST['Start_Date'] ) && isset( $_POST['End_Date'] ) )
```
Comment (pkgodara) [inline includes/AppLeave.php:90]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed
* Expected 1 space after IF keyword; 0 found
* Expected 0 spaces after opening bracket; 1 found
* Space after opening parenthesis of function call prohibited
* Expected 0 spaces before closing bracket; 1 found
* Expected 1 space after closing parenthesis; found 5

File: includes/AppLeave.php [hunk 14]
```diff
@@ -75,6 +75,27 @@
...
+	{
```
Comment (pkgodara) [inline includes/AppLeave.php:91]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 15]
```diff
@@ -75,6 +75,27 @@
...
+		if( strtotime($_POST['Start_Date']) > strtotime($_POST['End_Date']) )
```
Comment (pkgodara) [inline includes/AppLeave.php:92]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed
* Expected 1 space after IF keyword; 0 found
* Expected 0 spaces after opening bracket; 1 found
* Expected 0 spaces before closing bracket; 1 found
* Expected 1 space after closing parenthesis; found 9

File: includes/AppLeave.php [hunk 16]
```diff
@@ -75,6 +75,27 @@
...
+		{
```
Comment (pkgodara) [inline includes/AppLeave.php:93]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 17]
```diff
@@ -75,6 +75,27 @@
...
+			echo "ERROR: End date must not be less than Start date";
```
Comment (pkgodara) [inline includes/AppLeave.php:94]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 18]
```diff
@@ -75,6 +75,27 @@
...
+			die();
```
Comment (pkgodara) [inline includes/AppLeave.php:95]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 19]
```diff
@@ -75,6 +75,27 @@
...
+		}
```
Comment (pkgodara) [inline includes/AppLeave.php:96]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 20]
```diff
@@ -75,6 +75,27 @@
...
+	}
```
Comment (pkgodara) [inline includes/AppLeave.php:97]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/AppLeave.php [hunk 21]
```diff
@@ -75,6 +75,27 @@
...
+
```
Comment (pkgodara) [inline includes/AppLeave.php:98]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed

File: includes/application/application.php [hunk 22]
```diff
@@ -47,10 +47,10 @@
 }
 
 if( $flagSD )
-	echo "<tr><th>Start Date :</th><td> <input type=\"date\" name=Start_Date placeholder=\"Date yyyy-mm-dd\" required/></td></tr>";
+	echo "<tr><th>Start Date :</th><td> <input type=\"date\" name=Start_Date id=Start_Date placeholder=\"Date yyyy-mm-dd\" required/></td></tr>";
```
Comment (pkgodara) [inline includes/application/application.php:50]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed
* Line exceeds 120 characters; contains 145 characters

File: includes/application/application.php [hunk 23]
```diff
@@ -47,10 +47,10 @@
...
 
 if( $flagED )
-	echo "<tr><th>End Date :</th><td> <input type=\"date\" name=End_Date placeholder=\"Date yyyy-mm-dd\" required/></td></tr>";
+	echo "<tr><th>End Date :</th><td> <input type=\"date\" name=End_Date id=End_Date placeholder=\"Date yyyy-mm-dd\" required/></td></tr>";
```
Comment (pkgodara) [inline includes/application/application.php:53]: __[code_sniffer]__
* Spaces must be used to indent lines; tabs are not allowed
* Line exceeds 120 characters; contains 139 characters

**Comment by pkgodara** [comment]
Click here if you want to create code_sniffer's auto-correct Pull Request
[Auto Correct by SideCI](https://sideci.com/gh/80422762/issues/pull_request/68/diff/0e338af112e2aa29ac76d26fcf741b2c2c2808b2?open_modal=1#code_sniffer)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-code_sniffer)

**Review by NabeelQaiser — APPROVED** [review]

**Pull request merged**